File: store-ui/src/app/services/models/order/order-response.ts
import {OrderItemResponse} from "./order-item-response";
import {UserDetailsResponse} from "../user-details/user-details-response";

export interface OrderResponse {
    id?: number;
    userDetails?: UserDetailsResponse;
    items?: OrderItemResponse[];
    additionalCosts?: Record<string, number>[];
    totalPrice?: number;
    createTime?: string;
    updateTime?: string;
    status?: string;
}

File: services/order/src/main/java/com/ecommerce/order/OrderResponse.java
package com.ecommerce.order;

import com.ecommerce.orderitem.OrderItemResponse;
import com.ecommerce.userdetails.UserDetailsResponse;
import lombok.Builder;

import java.math.BigDecimal;
import java.time.LocalDateTime;
import java.util.Map;
import java.util.Set;

@Builder
public record OrderResponse(
        Long id,
        UserDetailsResponse userDetails,
        Set<OrderItemResponse> items,
        Map<String, BigDecimal> additionalCosts,
        BigDecimal totalPrice,
        LocalDateTime createTime,
        LocalDateTime updateTime,
        String status
) {}